### PR TITLE
Add compose sample along with UI tests

### PR DESCRIPTION
**Pull request by narenkmanoharan**
Remove old simple sample.

### CODE REVIEW

File: magellan-flow-sample/src/androidTest/java/com/wealthfront/magellan/assertions/AssertShown.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.espresso.util.HumanReadables
+import kotlin.reflect.KClass
+import org.hamcrest.CoreMatchers.allOf
+import org.hamcrest.CoreMatchers.containsString
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.Matcher
+
+fun assertShown(action: AssertionBuilder.() -> Unit) {
```
Comment (ryanmoelter) [inline magellan-flow-sample/src/androidTest/java/com/wealthfront/magellan/assertions/AssertShown.kt:20]: Oh cool, this works out of the box with robolectric? Or is this actually used in espresso?

File: magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/Expedition.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.wealthfront.magellan.sample
+
+import android.content.Context
+import com.wealthfront.magellan.core.Journey
+import com.wealthfront.magellan.sample.databinding.ExpeditionBinding
+
+class Expedition : Journey<ExpeditionBinding>(ExpeditionBinding::inflate, ExpeditionBinding::container) {
```
Comment (ryanmoelter) [inline magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/Expedition.kt:7]: lol we're actually calling this `Expedition` 👍

File: magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/FirstJourney.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.wealthfront.magellan.sample
+
+import android.content.Context
+import com.wealthfront.magellan.core.Journey
+import com.wealthfront.magellan.sample.databinding.FirstJourneyBinding
+
+class FirstJourney(
+  private val secondJourney: () -> Unit
```
Comment (ryanmoelter) [inline magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/FirstJourney.kt:8]: nit: `goToSecondJourney`?

File: magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+class MainActivity : AppCompatActivity() {
+
+  private val expeditionProvider = ExpeditionProvider
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentScreen(expeditionProvider.expedition)
+  }
+
+  override fun onBackPressed() {
```
Comment (ryanmoelter) [inline magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/MainActivity.kt:16]: Aww we can't do this automatically, can we?

File: magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/SecondJourney.kt [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package com.wealthfront.magellan.sample
+
+import android.content.Context
+import com.wealthfront.magellan.core.Journey
+import com.wealthfront.magellan.sample.databinding.SecondJourneyBinding
+
+class SecondJourney : Journey<SecondJourneyBinding>(SecondJourneyBinding::inflate, SecondJourneyBinding::container) {
```
Comment (ryanmoelter) [inline magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/SecondJourney.kt:7]: I get that we want to model what a second Journey would be like, but this Journey is useless and should just be the DetailScreen inside it. We should simplify this if you're not planning to add more to it

File: settings.gradle [hunk 6]
```diff
@@ -2,6 +2,6 @@ include ':magellan-library'
 include ':magellan-legacy'
 include ':magellan-rx'
 include ':magellan-rx2'
-include ':magellan-sample'
+include ':magellan-flow-sample'
```
Comment (ryanmoelter) [inline settings.gradle:5]: nit: Probably want this to be `magellan-journey-sample`

File: magellan-flow-sample/src/androidTest/java/com/wealthfront/magellan/assertions/AssertShown.kt (hunk 1 above)
Comment (narenkmanoharan) [reply magellan-flow-sample/src/androidTest/java/com/wealthfront/magellan/assertions/AssertShown.kt:20]: im just using this with espresso for now. But we can make a similar version that works with robolectric, i think

File: magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/Expedition.kt (hunk 2 above)
Comment (narenkmanoharan) [reply magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/Expedition.kt:7]: yup 😛

File: magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/MainActivity.kt (hunk 4 above)
Comment (narenkmanoharan) [reply magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/MainActivity.kt:16]: unfortunely, no :(

File: magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/SecondJourney.kt (hunk 5 above)
Comment (narenkmanoharan) [reply magellan-flow-sample/src/main/java/com/wealthfront/magellan/sample/SecondJourney.kt:7]: I am planning to show off some history rewrite functionality and flow exit stuff in the future.

**Pull request merged**